Build failure: beszel

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

```
nix flake update
sudo darwin-rebuild switch --flake .
```

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: Cannot build '/nix/store/g2nffgjhgyi1nkzf95si2424zpnxx9sj-beszel-0.18.6.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/7fzhcdi5lnqcd0yqbi563w5nwh7y0yvk-beszel-0.18.6
       Last 25 log lines:
       > 2026/04/15 08:17:28 WARN Data directory not found
       > 2026/04/15 08:17:28 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
       > 2026/04/15 08:17:28 INFO Starting SSH server addr=:45987 network=tcp
       > 2026/04/15 08:17:28 INFO SSH connected addr=127.0.0.1:52053
       > 2026/04/15 08:17:28 WARN Data directory not found
       > 2026/04/15 08:17:28 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
       > 2026/04/15 08:17:28 WARN Data directory not found
       > 2026/04/15 08:17:28 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
       > 2026/04/15 08:17:28 WARN Data directory not found
       > 2026/04/15 08:17:28 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
       > 2026/04/15 08:17:28 WARN Data directory not found
       > 2026/04/15 08:17:28 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
       > 2026/04/15 08:17:28 WARN Data directory not found
       > 2026/04/15 08:17:28 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
       > 2026/04/15 08:17:28 WARN Data directory not found
       > 2026/04/15 08:17:28 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
       > 2026/04/15 08:17:28 WARN Data directory not found
       > 2026/04/15 08:17:28 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
       > 2026/04/15 08:17:28 INFO SMART_DEVICES value=/dev/sda:jmb39x-q,0|/dev/nvme0:nvme
       > 2026/04/15 08:17:28 INFO SMART_DEVICES value="/dev/sda:sat, /dev/nvme0:nvme"
       > 2026/04/15 08:17:28 INFO SMART_DEVICES value=:sat
       > 2026/04/15 08:17:28 INFO SMART_DEVICES value="   "
       > FAIL
       > FAIL        github.com/henrygd/beszel/agent 2.869s
       > FAIL
       For full logs, run:
         nix log /nix/store/g2nffgjhgyi1nkzf95si2424zpnxx9sj-beszel-0.18.6.drv
error: Cannot build '/nix/store/3wrydb4gb943ln298l5wyxn3rrr7r9lw-beszel-0.18.6-fish-completions.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/fcdw6mvlchlxlcjcl8fm5cz4pg3d0w4d-beszel-0.18.6-fish-completions
error: Cannot build '/nix/store/35fyiv1zcdpxaggd99jhiy6yww51basq-beszel-agent-start.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/m3vmy87526vxp2fgq4rxqqn6xddgikmk-beszel-agent-start
error: Cannot build '/nix/store/mlwis4pd329wwwcbz44paip04p4712rk-home-manager-applications.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/dhrrdn6abc4pz9hjs256rhgma90jlf41-home-manager-applications
error: Cannot build '/nix/store/19dhs6gmn9bnsbg8fxr02yc6lmqqjd5k-home-manager-fonts.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/8qvvx12d1pm3skyb3jwa9wy5iwsagiir-home-manager-fonts
error: Cannot build '/nix/store/sch95410pkwa8gzgv96dry4ngh0vnv1j-home-manager-path.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/wrkyw6ah0i9ys9sfpj75l3i43xl6659j-home-manager-path
error: Cannot build '/nix/store/y6fyfqfks6sgla9lzhry6n6swy8ippwj-man-paths.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/1n9nmnci506gqpsm2yrj5qrl213ws18r-man-paths
error: Build failed due to failed dependency
error: Build failed due to failed dependency
error: Build failed due to failed dependency
error: Build failed due to failed dependency
error: Build failed due to failed dependency
error: Build failed due to failed dependency
error: Cannot build '/nix/store/mjkhcfc9mlyn7d0s5gd1xqqna0s4i66l-activation-script.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/y8s9kd91zcqs1y6dixr9502306j561rx-activation-script
error: Cannot build '/nix/store/c7r4ysb3jvv4pfiqv78jj70p7kshrffr-hm_LibraryFonts.homemanagerfontsversion.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/8asp0hl5dmbp09zv26901ncx7lwww6ms-hm_LibraryFonts.homemanagerfontsversion
error: Cannot build '/nix/store/a030x9v42431pskycjqj5c5nrmbnifps-home-manager-generation.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/13iqi41zx62539q13bnkbfn97bp7hsb1-home-manager-generation
error: Cannot build '/nix/store/4k43c8d8gsgzwm2cpd9avb7mfv52nw6j-man-cache.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/7w47bvais5cb0gb32chk0gvsb9y460f8-man-cache
error: Cannot build '/nix/store/96pmli9c3gv4dimw8ahmlqd8xlj593bk-nixdwn.qi.beszel-agent.plist.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/0v1nymh8xg3r5qfzi56gqfn4gvy8hrql-nixdwn.qi.beszel-agent.plist
error: Cannot build '/nix/store/40mk82zw67brd3h9i4j2hm6k3035qjhh-qi-fish-completions.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/pwsnfvzhs644q1mfz3f0mq06d79crzj3-qi-fish-completions
error: Cannot build '/nix/store/7nqls61w6mpy3ny1dfhv1qwhrihpkw3j-user-environment.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/6d8ph96ix1ssrdyd6nyhp5773d0igqx6-user-environment
error: Build failed due to failed dependency
error: Build failed due to failed dependency
error: Build failed due to failed dependency
error: Cannot build '/nix/store/ripj9kwspwl6dv8b6905b481mg5m21si-activation-qi.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/j9smxaiq2cfkg5zsdb3wmdy67cfn4d34-activation-qi
error: Cannot build '/nix/store/2zxh5n8m6dhd26a9a0l0l8h57ncnpmxr-etc.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/smqqjm7kmdi9zzm9r7yykwabq6d0ab94-etc
error: Cannot build '/nix/store/d2psi5g52ylb0sxl4lv32zfidpga2hcp-launchd.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/3qd5hgwpy7s4r3ckayqj5fry88r9m8ir-launchd
error: Build failed due to failed dependency
error: Cannot build '/nix/store/h7pzgsfg4c28r0zszm1hsn1m9cvpqs52-darwin-system-26.05.06648f4.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/kasvvbclj95zzpx26l8zq9skf8y9icn3-darwin-system-26.05.06648f4
```

### Additional context

version: `0.18.4` build success on this machine

### System metadata

```
 - system: `"aarch64-darwin"`
 - host os: `Darwin 25.4.0, macOS 26.4.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Determinate Nix 3.17.0) 2.33.3`
 - nixpkgs: `/nix/store/f9zw9nsp96n5zlfr2cwl8msrd1mfbzm1-source`
```

### Notify maintainers


@Bot-wxt1221 @arunoruto @BonusPlay 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

<details>
  <summary>
    You cut the logs where it gets interesting and it should look like this:
  </summary>

```log
beszel> Running phase: checkPhase
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestNewWebSocketClient4289130563/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestWebSocketClient_GetOptions1489619190/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestWebSocketClient_VerifySignature1955169968/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestWebSocketClient_HandleHubRequest2486870479/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestWebSocketClient_Close1548112121/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestWebSocketClient_ConnectRateLimit2949119231/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestConnectionManager_NewConnectionManager2934327400/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestConnectionManager_StateTransitions1883190461/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO WebSocket connected host=localhost:8080
beszel> 2026/04/15 23:21:17 INFO SSH connection established
beszel> 2026/04/15 23:21:17 WARN Disconnected from hub
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestConnectionManager_EventHandling1922136100/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO WebSocket connected host=localhost:8080
beszel> 2026/04/15 23:21:17 INFO SSH connection established
beszel> 2026/04/15 23:21:17 WARN Disconnected from hub
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestConnectionManager_TickerManagement4231254621/001
beszel> 2026/04/15 23:21:17 WARN WebSocket connection failed err="dial tcp [::1]:8080: connect: connection refused"
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestConnectionManager_WebSocketConnectionFlow784390224/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestConnectionManager_ReconnectionLogic1679659540/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 WARN Disconnected from hub
beszel> 2026/04/15 23:21:17 INFO Starting SSH server addr="" network=""
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestConnectionManager_ConnectWithRateLimit594468699/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 WARN WebSocket connection failed err="dial tcp [::1]:8080: connect: connection refused"
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestConnectionManager_StartWithInvalidConfig2385818699/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestConnectionManager_CloseWebSocket1336699271/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Data directory path=/nix/var/nix/builds/nix-61242-721484610/TestConnectionManager_ConnectFlow1945077208/001
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:17 INFO Starting SSH server addr="" network=""
beszel> 2026/04/15 23:21:17 INFO Using disk I/O fallback requested=ada0p2 selected=ada0
beszel> 2026/04/15 23:21:17 INFO Using disk I/O fallback requested=nvme0n1p2 selected=nvme0n1
beszel> 2026/04/15 23:21:17 INFO Using disk I/O fallback requested=ada0p2 selected=ada0
beszel> 2026/04/15 23:21:17 INFO Detected disk name=ada0 device=/dev/ada0p2 mount=/ io=ada0 root=true
beszel> 2026/04/15 23:21:17 WARN Partition details not found filesystem=missing-disk
beszel> 2026/04/15 23:21:17 INFO Using disk I/O fallback requested=nvme0n1p2 selected=nvme0n1
beszel> 2026/04/15 23:21:17 INFO Detected disk name=nvme0n1 device=nvme0n1 mount=/ io=nvme0n1 root=true
beszel> 2026/04/15 23:21:17 WARN Using most active device for root I/O; set FILESYSTEM to override device=sda
beszel> 2026/04/15 23:21:17 WARN Root I/O device not detected; set FILESYSTEM to override
beszel> 2026/04/15 23:21:17 INFO Detected disk name=backup device=/dev/sdb1 mount=/mnt/backup io=sdb1 root=false
beszel> 2026/04/15 23:21:17 INFO Detected disk name=archive device=archive mount=/srv/archive io=archive root=false
beszel> 2026/04/15 23:21:17 ERROR Invalid filesystem name=/missing/archive err="file does not exist"
beszel> 2026/04/15 23:21:17 INFO Detected disk name=fast device=/dev/sda1 mount=/mnt/fast io=sda1 root=false
beszel> 2026/04/15 23:21:17 INFO Detected disk name=cold device=archive mount=/srv/archive io=archive root=false
beszel> 2026/04/15 23:21:17 INFO Detected disk name=Archive device=newdisk mount=/extra-filesystems/newdisk__Archive io=newdisk root=false
beszel> 2026/04/15 23:21:17 INFO Using disk I/O fallback requested=nvme0n1p2 selected=nvme0n1
beszel> 2026/04/15 23:21:17 INFO Using disk I/O fallback requested=sd selected=sda
beszel> 2026/04/15 23:21:17 INFO Using disk I/O fallback requested=sd selected=sda
beszel> 2026/04/15 23:21:17 INFO Using disk I/O fallback requested=nda0p2 selected=nda0
beszel> 2026/04/15 23:21:17 INFO Using disk I/O fallback requested=sd selected=sda
beszel> 2026/04/15 23:21:17 ERROR Error getting disk stats name=/mnt/storage err="no such file or directory"
beszel> 2026/04/15 23:21:17 ERROR Error getting disk stats name=/mnt/storage err="no such file or directory"
beszel> 2026/04/15 23:21:17 ERROR Error getting disk stats name=/mnt/storage err="no such file or directory"
beszel> 2026/04/15 23:21:17 WARN Device not found in diskstats name=sda
beszel> 2026/04/15 23:21:17 INFO Docker 24.0.7 is outdated. Upgrade if possible. See https://github.com/henrygd/beszel/issues/58
beszel> 2026/04/15 23:21:17 INFO Docker 24.0.7 is outdated. Upgrade if possible. See https://github.com/henrygd/beszel/issues/58
beszel> 2026/04/15 23:21:17 WARN Bad network delta container=test
beszel> 2026/04/15 23:21:17 WARN Bad network delta container=test
beszel> 2026/04/15 23:21:17 WARN Bad network delta container=test
beszel> 2026/04/15 23:21:17 WARN Ignoring unknown GPU collector collector=bad
beszel> --- FAIL: TestCollectorStartHelpers (0.20s)
beszel>     --- FAIL: TestCollectorStartHelpers/nvidia-smi_collector (0.05s)
beszel>         gpu_test.go:1260:
beszel>                 Error Trace:    /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1260
beszel>                                                         /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1361
beszel>                 Error:          Should be true
beszel>                 Test:           TestCollectorStartHelpers/nvidia-smi_collector
beszel>     --- FAIL: TestCollectorStartHelpers/rocm-smi_collector (0.05s)
beszel>         gpu_test.go:1282:
beszel>                 Error Trace:    /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1282
beszel>                                                         /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1361
beszel>                 Error:          Should be true
beszel>                 Test:           TestCollectorStartHelpers/rocm-smi_collector
beszel>     --- FAIL: TestCollectorStartHelpers/tegrastats_collector (0.05s)
beszel>         gpu_test.go:1306:
beszel>                 Error Trace:    /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1306
beszel>                                                         /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1361
beszel>                 Error:          Max difference between 70 and 0 allowed is 0.1, but difference was 70
beszel>                 Test:           TestCollectorStartHelpers/tegrastats_collector
beszel>     --- FAIL: TestCollectorStartHelpers/nvtop_collector (0.05s)
beszel>         gpu_test.go:1329:
beszel>                 Error Trace:    /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1329
beszel>                                                         /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1361
beszel>                 Error:          Should be true
beszel>                 Test:           TestCollectorStartHelpers/nvtop_collector
beszel> 2026/04/15 23:21:17 WARN nvidia-smi failed, restarting err="exit status 127"
beszel> 2026/04/15 23:21:17 WARN Error collecting GPU data via nvtop err="no valid GPU data found"
beszel> 2026/04/15 23:21:17 WARN tegrastats failed, restarting err="exit status 127"
beszel> 2026/04/15 23:21:17 WARN Error collecting GPU data via nvtop err="invalid character 'o' in literal null (expecting 'u')"
beszel> --- FAIL: TestNewGPUManagerPriorityNvtopFallback (0.15s)
beszel>     gpu_test.go:1387:
beszel>                 Error Trace:    /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1387
beszel>                 Error:          Should be true
beszel>                 Test:           TestNewGPUManagerPriorityNvtopFallback
beszel> 2026/04/15 23:21:17 WARN rocm-smi is deprecated and may be removed in a future release
beszel> 2026/04/15 23:21:17 WARN nvidia-smi failed, restarting err="exit status 127"
beszel> --- FAIL: TestNewGPUManagerPriorityMixedCollectors (0.15s)
beszel>     gpu_test.go:1419:
beszel>                 Error Trace:    /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1419
beszel>                 Error:          Should be true
beszel>                 Test:           TestNewGPUManagerPriorityMixedCollectors
beszel>     gpu_test.go:1420:
beszel>                 Error Trace:    /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1420
beszel>                 Error:          Should be true
beszel>                 Test:           TestNewGPUManagerPriorityMixedCollectors
beszel> 2026/04/15 23:21:17 WARN Failed to initialize NVML err="nvml not supported on this platform"
beszel> 2026/04/15 23:21:17 WARN Error collecting AMD GPU data via rocm-smi err="exit status 127"
beszel> 2026/04/15 23:21:17 WARN Error collecting Intel GPU data; see https://beszel.dev/guide/gpu err="no valid GPU data found"
beszel> --- FAIL: TestNewGPUManagerPriorityNvmlFallbackToNvidiaSmi (0.15s)
beszel>     gpu_test.go:1439:
beszel>                 Error Trace:    /nix/var/nix/builds/nix-61242-721484610/source/agent/gpu_test.go:1439
beszel>                 Error:          Should be true
beszel>                 Test:           TestNewGPUManagerPriorityNvmlFallbackToNvidiaSmi
beszel> 2026/04/15 23:21:17 WARN Ignoring unknown GPU collector collector=bad
beszel> 2026/04/15 23:21:17 WARN Ignoring unknown GPU collector collector=unknown
beszel> 2026/04/15 23:21:17 WARN Failed to initialize NVML err="nvml not supported on this platform"
beszel> 2026/04/15 23:21:17 WARN nvidia-smi failed, restarting err="exit status 127"
beszel> 2026/04/15 23:21:17 WARN tegrastats failed, restarting err="exit status 127"
beszel> 2026/04/15 23:21:18 WARN Invalid net stats. Resetting. sent=11000000000 recv=1000000
beszel> 2026/04/15 23:21:18 WARN Invalid net stats. Resetting. sent=1000000 recv=11000000000
beszel> 2026/04/15 23:21:18 WARN Invalid net stats. Resetting. sent=12000000000 recv=13000000000
beszel> 2026/04/15 23:21:18 INFO SYS_SENSORS path=/custom/path
beszel> 2026/04/15 23:21:18 INFO SYS_SENSORS path=/test/path
beszel> 2026/04/15 23:21:18 WARN Error updating temperatures err="temperature collection timed out"
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 INFO Starting SSH server addr=:45987 network=tcp
beszel> 2026/04/15 23:21:18 INFO SSH connected addr=127.0.0.1:58709
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 INFO Starting SSH server addr=127.0.0.1:45988 network=tcp4
beszel> 2026/04/15 23:21:18 INFO SSH connected addr=127.0.0.1:58710
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 INFO Starting SSH server addr=[::1]:45989 network=tcp6
beszel> 2026/04/15 23:21:18 INFO SSH connected addr=[::1]:58711
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 INFO Starting SSH server addr=/nix/var/nix/builds/nix-61242-721484610/TestStartServer2000946701/001/beszel-test.sock network=unix
beszel> 2026/04/15 23:21:18 INFO SSH connected addr=""
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 INFO Starting SSH server addr=:45987 network=tcp
beszel> 2026/04/15 23:21:18 WARN Invalid SSH key addr=127.0.0.1:58712
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 INFO Starting SSH server addr=:45987 network=tcp
beszel> 2026/04/15 23:21:18 INFO SSH connected addr=127.0.0.1:58713
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 WARN Data directory not found
beszel> 2026/04/15 23:21:18 WARN Using most active device for root I/O; set FILESYSTEM to override device=disk0
beszel> 2026/04/15 23:21:18 INFO SMART_DEVICES value=/dev/sda:jmb39x-q,0|/dev/nvme0:nvme
beszel> 2026/04/15 23:21:18 INFO SMART_DEVICES value="/dev/sda:sat, /dev/nvme0:nvme"
beszel> 2026/04/15 23:21:18 INFO SMART_DEVICES value=:sat
beszel> 2026/04/15 23:21:18 INFO SMART_DEVICES value="   "
beszel> FAIL
beszel> FAIL    github.com/henrygd/beszel/agent 1.986s
beszel> FAIL
```
</details>

We can already see, that the `TestCollectorStartHelpers/nvtop_collector` test is being skipped. I guess since nvidia isn't free and would need to be approved to be run on the servers. The other binaries can be found in nixpkgs and will be resolved on demand?
On Mac thats a different story, since no package is available there. Hence, we skip a few more packages:
```
        "TestCollectorStartHelpers"
        "TestNewGPUManagerPriorityNvtopFallback"
        "TestNewGPUManagerPriorityMixedCollectors"
        "TestNewGPUManagerPriorityNvmlFallbackToNvidiaSmi"
        "TestNewGPUManagerConfiguredCollectorsMustStart"
        "TestNewGPUManagerConfiguredNvmlBypassesCapabilityGate"
        "TestNewGPUManagerJetsonIgnoresCollectorConfig"
```
Thats also what the log lines are telling us...

When this is fixed, another error appears:
```log
beszel> Running phase: checkPhase
beszel> ok      github.com/henrygd/beszel/agent 1.782s
beszel> ok      github.com/henrygd/beszel/agent/deltatracker    0.328s
beszel> ok      github.com/henrygd/beszel/agent/health  0.357s
beszel> ok      github.com/henrygd/beszel/agent/utils   0.331s
beszel> panic: runtime error: invalid memory address or nil pointer dereference
beszel> [signal SIGSEGV: segmentation violation code=0x2 addr=0xb0 pc=0x101467140]
beszel>
beszel> goroutine 38652 [running]:
beszel> github.com/pocketbase/pocketbase/core.(*BaseApp).RecordQuery(0x6d7eee754008, {0x102a41b40?, 0x6d7eee446008?})
beszel>         /nix/var/nix/builds/nix-71066-3105078627/source/vendor/github.com/pocketbase/pocketbase/core/record_query.go:38 +0x70
beszel> github.com/pocketbase/pocketbase/core.(*BaseApp).FindRecordById(0x6d7eee754008, {0x1028b7480?, 0x102a76260?}, {0x6d7eedefeee0, 0xf}, {0x0, 0x0, 0x1015f1d58?})
beszel>         /nix/var/nix/builds/nix-71066-3105078627/source/vendor/github.com/pocketbase/pocketbase/core/record_query.go:223 +0x78
beszel> github.com/henrygd/beszel/internal/hub/systems.(*System).getRecord(0x6d7eee63f0a0)
beszel>         /nix/var/nix/builds/nix-71066-3105078627/source/internal/hub/systems/system.go:347 +0x50
beszel> github.com/henrygd/beszel/internal/hub/systems.(*System).setDown(0x6d7eee63f0a0, {0x102a7d940, 0x6d7eee692550})
beszel>         /nix/var/nix/builds/nix-71066-3105078627/source/internal/hub/systems/system.go:362 +0x94
beszel> github.com/henrygd/beszel/internal/hub/systems.(*System).StartUpdater(0x6d7eee63f0a0)
beszel>         /nix/var/nix/builds/nix-71066-3105078627/source/internal/hub/systems/system.go:88 +0xc4
beszel> created by github.com/henrygd/beszel/internal/hub/systems.(*SystemManager).AddSystem in goroutine 37891
beszel>         /nix/var/nix/builds/nix-71066-3105078627/source/internal/hub/systems/system_manager.go:253 +0x17c
beszel> FAIL    github.com/henrygd/beszel/internal/alerts       63.711s
beszel> FAIL
```

The culprit is this line of code: https://github.com/henrygd/beszel/blob/e5507fa1060ba3d97a9e6ba380db1fe5db401ef0/internal/hub/systems/system.go#L66
It hangs, so we need to tell him to not freak out with something like this:
```nix
  postPatch = ''
    substituteInPlace internal/hub/systems/system.go \
      --replace-fail "func (sys *System) StartUpdater() {" "func (sys *System) StartUpdater() { defer func() { recover() }();"
  '';
```

While it builds now, I am not sure if it will work on MacOS! I don't have a hub at home to connect to. I will have to try it out at work at some point.

@handy-sun would you be comfortable to try patching beszel on your system and see if it works?

--- a/pkgs/by-name/be/beszel/package.nix
+++ b/pkgs/by-name/be/beszel/package.nix
@@ -53,8 +53,6 @@ buildGo126Module (finalAttrs: {
 
   vendorHash = "sha256-TVpZbK9V9/GqpVFcjF7QGD5XJJHzRgjVXZOImHQTR1k=";
 
-  tags = [ "testing" ];
-
   preBuild = ''
     mkdir -p internal/site/dist
     cp -r ${finalAttrs.webui}/* internal/site/dist
@@ -68,7 +66,10 @@ buildGo126Module (finalAttrs: {
         "TestConfigSyncWithTokens"
       ];
     in
-    [ "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$" ];
+    [
+      "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$"
+      "-tags=testing"
+    ];
 
   postInstall = ''
     mv $out/bin/agent $out/bin/beszel-agent

--- a/pkgs/by-name/be/beszel/package.nix
+++ b/pkgs/by-name/be/beszel/package.nix
@@ -1,7 +1,9 @@
 {
+  stdenv,
   buildGo126Module,
   lib,
   fetchFromGitHub,
+  fetchpatch,
   nix-update-script,
   buildNpmPackage,
   nixosTests,
@@ -53,6 +55,15 @@ buildGo126Module (finalAttrs: {
 
   vendorHash = "sha256-TVpZbK9V9/GqpVFcjF7QGD5XJJHzRgjVXZOImHQTR1k=";
 
+  patches = [
+    # https://github.com/NixOS/nixpkgs/pull/513197
+    (fetchpatch {
+      name = "fix-updater-after-system-manager-shutdown.patch";
+      url = "https://github.com/henrygd/beszel/commit/c538d1de1cf3f4664a2d98086341884a217846e7.patch";
+      hash = "sha256-voIT9b14pgfhnbJrqgoIbQtwZPU1JF0fblybjG9mzvM=";
+    })
+  ];
+
   preBuild = ''
     mkdir -p internal/site/dist
     cp -r ${finalAttrs.webui}/* internal/site/dist
@@ -64,6 +75,19 @@ buildGo126Module (finalAttrs: {
         "TestCollectorStartHelpers/nvtop_collector"
         "TestApiRoutesAuthentication/GET_/update_-_shouldn't_exist_without_CHECK_UPDATES_env_var"
         "TestConfigSyncWithTokens"
+        # This subtest assumes enough host CPUs for an 8s CPU delta over 1s to stay below 100%.
+        "TestServiceUpdateCPUPercent/subsequent_call_calculates_CPU_percentage"
+      ]
+      ++ lib.optionals stdenv.hostPlatform.isDarwin [
+        "TestCollectorStartHelpers/nvidia-smi_collector"
+        "TestCollectorStartHelpers/rocm-smi_collector"
+        "TestCollectorStartHelpers/tegrastats_collector"
+        "TestNewGPUManagerPriorityNvtopFallback"
+        "TestNewGPUManagerPriorityMixedCollectors"
+        "TestNewGPUManagerPriorityNvmlFallbackToNvidiaSmi"
+        "TestNewGPUManagerConfiguredCollectorsMustStart"
+        "TestNewGPUManagerConfiguredNvmlBypassesCapabilityGate"
+        "TestNewGPUManagerJetsonIgnoresCollectorConfig"
       ];
     in
     [
@@ -75,6 +99,8 @@ buildGo126Module (finalAttrs: {
     mv $out/bin/agent $out/bin/beszel-agent
     mv $out/bin/hub $out/bin/beszel-hub
   '';
+
+  __darwinAllowLocalNetworking = true;
 
   passthru = {
     updateScript = nix-update-script {

